numpy/core/src/umath/loops.c.src

```diff
@@ -1126,11 +1126,49 @@ NPY_NO_EXPORT void
     BINARY_LOOP {
         const @type@ in1 = *(@type@ *)ip1;
         const @type@ in2 = *(@type@ *)ip2;
-        *((npy_bool *)op1) = in1 @OP@ in2;
+        const npy_bool res = in1 @OP@ in2;
+        *((npy_bool *)op1) = res;
+
+        if ((in1 == NPY_DATETIME_NAT || in2 == NPY_DATETIME_NAT) && res) {
+            NPY_ALLOW_C_API_DEF
+            NPY_ALLOW_C_API;
+            /* 2016-01-18, 1.11 */
+            if (DEPRECATE_FUTUREWARNING(
```

Comment:
My understanding to the differences is this. A `DeprecationWarning` means that in the future either the feature will be removed or an exception will be raises. OTOH, a `FutureWarning` means that the feature will remain the its behavior will change.

I suppose the macro name `DEPRECATE_FUTUREWARNING` is a bit misleading, it just issues a `FutureWarning`.